src/software/geom/spline.cpp

```diff
@@ -0,0 +1,55 @@
+#include "software/geom/spline.h"
+
+Spline::Spline(const std::vector<Point>& points)
+{
+    initLinearSegments(points);
+}
+
+Spline::Spline(const std::initializer_list<Point>& points)
+{
+    initLinearSegments(points);
+}
+
+Point Spline::calculateValue(double val) const
+{
+    if (val < 0)
+    {
+        return start_point;
+    }
+
+    if (val >= segments.size())
+    {
+        return end_point;
```

Comment:
hmm, might be more meaningful to return `std::nullopt` here. If people want the endpoint we should have a specific method for that.

Reply:
so a spline is only defined from start point to end point and any other input is invalid?

Reply:
I think so? It doesn't "make sense" to try access something outside a finite spline / curve.

Reply:
ok

Reply:
Yeah, if you have a map from `[0,1]` -> R^2, asking for the point at `100` doesn't really make sense.

Reply:
throwing exception here, please review